ui/src/app/applications/components/utils.tsx

```diff
@@ -289,8 +295,8 @@ export const deletePopup = async (ctx: ContextApis, resource: ResourceTreeNode,
                     resourceName: vals.resourceName !== resource.name && 'Enter the resource name to confirm the deletion'
                 },
             submit: async (vals, _, close) => {
-                const force = deleteOptions.option === 'force';
-                const orphan = deleteOptions.option === 'orphan';
+                const force = vals.propagationPolicy === 'force';
+                const orphan = vals.propagationPolicy === 'non-cascading (orphan)';
```

Comment:
I think the values that should be set should be 'internal strings' or keys, and not UI strings that are exposed.  These can change, and if these strings were translated, then we have to update this.

It looks like the application delete code is like this this too.

Reply:
Hi @chetan-rns , I suppose the suggested change can be done in a separate PR since it affects application delete too.

Reply:
Thanks for the review @keithchong. I have added the suggested changes for both app and resource deletion

Reply:
Updated the PR title since it involved refactoring